src/backend/distributed/utils/colocation_utils.c

```diff
@@ -728,12 +719,47 @@ CheckDistributionColumnType(Oid sourceRelationId, Oid targetRelationId)
 }
 
 
+/*
+ * GetColumnTypeEquality checks if distribution column types and collations
+ * of the given columns are same. The function sets the boolean pointers.
+ */
+void
+GetColumnTypeEquality(Var *sourceDistributionColumn, Var *targetDistributionColumn,
+					  bool *columnTypesSame, bool *columnCollationsSame)
+{
```

Comment:
this part is just a refactor of `CheckDistributionColumnType ` as we only need `GetColumnTypeEquality` in  `citus_internal_add_partition_metadata`